analysis/organize_data.qmd

```diff
@@ -47,188 +47,197 @@ gss <- gss |>
+      god==3 ~ "higher power",
+      god>=4 & god<=5 ~ "believe with doubts",
+      god==6 ~ "believe with no doubts",
+      TRUE ~ NA_character_),
+    levels=c("no belief","higher power", "believe with doubts",
+             "believe with no doubts")))
 
-table(gss$sex, gss$ssex, exclude=NULL)
+table(gss$god, gss$believe_god, exclude=NULL)
 ```
 
 ```{r}
-#| label: Reordering reliten variable
+#| label: code-relig-imp
 gss <- gss |>
-  mutate(religimp=case_when(
+  mutate(relig_imp=case_when(
     reliten==1 ~ 4, # strong
-    reliten==2 ~ 3, # somewhat strong
-    reliten==3 ~ 2, # not strong
+    reliten==3 ~ 3, # somewhat strong
+    reliten==2 ~ 2, # not very strong
     reliten==4 ~ 1, # no religion
   ))
 
-table(gss$reliten, gss$religimp, exclude=NULL)
+table(gss$reliten, gss$relig_imp, exclude=NULL)
 
 ```
 
 ```{r}
-#| label: Recoding reliten variable
+#| label: code-relig-imp-categorical
 gss <- gss |>
-  mutate(religimp=case_when(
-    reliten==4 ~ "strong",
-    reliten==3 ~ "somewhat strong",
-    reliten==2 ~ "not strong",
-    reliten==1 ~ "no religion",
-    TRUE ~ NA_character_
-  ))
+  mutate(relig_imp_cat=factor(
+    case_when(
+      relig_imp==4 ~ "strong",
+      relig_imp==3 ~ "somewhat strong",
+      relig_imp==2 ~ "not strong",
+      relig_imp==1 ~ "no religion",
+      TRUE ~ NA_character_),
+    levels=c("no religion","not strong","somewhat strong","strong")))
 
-table(gss$reliten, gss$religimp, exclude=NULL)
+table(gss$relig_imp, gss$relig_imp_cat, exclude=NULL)
 ```
 
 ```{r}
-#| label: reverse coding prayer frequency
-gss$pray <- 7-gss$pray
+#| label: code-reverse-pray
 ```
 
 ```{r}
-#| label: verifying alpha
-alpha(gss[, c("attend", "pray", "god", "reliten")])
+#| label: religiosity-alpha
+alpha(gss[, c("attend", "pray", "god", "relig_imp")])
 ```
 
 ```{r}
-#| label: Standardizing belief in relgiosity variables.
+#| label: create-summated-scale
 standardize <- function(x) {
   (x-mean(x, na.rm=TRUE))/sd(x, na.rm=TRUE)
 }
 gss <- gss |>
   mutate(religiosity=standardize(standardize(attend)+
-                                   standardize(pray)+standardize(god)+
-                                   standardize(reliten)))
+                                   standardize(pray)+
+                                   standardize(god)+
+                                   standardize(relig_imp)))
+
+# check correlations with components
+cor(gss[,c("religiosity","attend", "pray", "god", "relig_imp")],
+    use="pairwise.complete.obs")
+
+# check missing value pattern by year
+table(gss$year, is.na(gss$religiosity))
 ```
 
 ## Save it all
 
 ```{r}
 #| label: save-analytical-data
 gss <- gss |>
-  select(year, edattain, ssex, religattend,
-         prayfreq, racecat, religiosity, religpref, religimp, bgod)
+  select(year, resp_sex, educ_attain, race,
+         relig_affil, religiosity, relig_attend, pray_freq, believe_god, 
+         relig_imp_cat) |>
+  rename(relig_imp=relig_imp_cat) |>
+  drop_na()
```

Comment:
I am currently just dropping all the missing values at this point. I think for your project its a reasonable number of cases too lose, but you could also consider an imputation strategy.